keyboards/rgbkb/sol3/keymaps/default/keymap.c

```diff
@@ -0,0 +1,246 @@
+            break;
+        case LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER: {
+            rgb_matrix_set_flags(LED_FLAG_UNDERGLOW);
+            rgb_matrix_set_color_all(0, 0, 0);
+            }
+            break;
+        case LED_FLAG_UNDERGLOW: {
+            rgb_matrix_set_flags(LED_FLAG_NONE);
+            rgb_matrix_disable_noeeprom();
+            }
+            break;
+        default: {
+            rgb_matrix_set_flags(LED_FLAG_ALL);
+            rgb_matrix_enable_noeeprom();
+            }
+            break;
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode)
+    {
+        case MENU_BTN:
+            if (record->event.pressed) {
+                rgb_menu_selection();
+            }
+            return false;
+        case MENU_UP:
+            if (record->event.pressed) {
+                rgb_menu_action(true);
+            }
+            return false;
+        case MENU_DN:
+            if (record->event.pressed) {
+                rgb_menu_action(false);
+            }
+            return false;
+        case RGB_RST:
+            if (record->event.pressed) {
+                eeconfig_update_rgb_matrix_default();
+            }
+            return false;
+        case RGB_TOG:
+            if (record->event.pressed) {
+                rgb_matrix_increase_flags();
+            }
+            return false;
+        case TCH_TOG:
+            if (record->event.pressed) {
+                touch_encoder_toggle();
+            }
+            return false;  // Skip all further processing of this key
+        default:
+            return true;
+    }
+}
+
+static void render_layer(void) {
+    // Host Keyboard Layer Status
+    oled_write_P(PSTR("Layer"), false);
+    switch (get_highest_layer(layer_state)) {
+        case _QWERTY:
+            oled_write_ln_P(PSTR("QWRTY"), false);
+            break;
+        case _COLEMAK:
+            oled_write_ln_P(PSTR("Colemk"), false);
+            break;
+        case _GAME:
+            oled_write_ln_P(PSTR("Game  "), false);
+            break;
+        case _FN:
+            oled_write_ln_P(PSTR("FN   "), false);
+            break;
+        case _ADJUST:
+            oled_write_ln_P(PSTR("Adjst"), false);
+            break;
+        default:
+            oled_write_ln_P(PSTR("Undef"), false);
+    }
+}
+
+static void render_leds(void)
+{
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUMLK")     : PSTR("     "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAPLK")    : PSTR("     "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCRLK")  : PSTR("     "), false);
+}
+
+static void render_touch(void)
+{
+    // Host Touch LED Status
+    oled_write_P(!touch_encoder_toggled() ? PSTR("TOUCH")  : PSTR("     "), false);
+    oled_write_P(touch_encoder_calibrating() ? PSTR("CLBRT")  : PSTR("     "), false);
+}
+
+bool oled_task_user(void) {
```

Comment:
Please move/add default oled behavior using `oled_render_kb` to the keyboard code.